crates/biome_service/src/file_handlers/mod.rs

```diff
@@ -1036,8 +1036,9 @@ impl<'a, 'b> LintVisitor<'a, 'b> {
 
         let path = self.path.expect("File path");
 
+        let is_recommended = R::METADATA.recommended;
         let recommended_enabled = self.settings.linter_recommended_enabled();
-        if !recommended_enabled {
+        if !is_recommended || !recommended_enabled {
```

Comment:
Need a opinion here; the domains enabled by the dependencies in `package.json` will enable **all** rules in the domain for now, is this intended? I think this "magic default" should only enable recommended rules here.

Reply:
Not intended, it's a bug. We should enable only the recommended rules for that domain

Reply:
Thank you for confirming, I'll keep the change and ship in the next minor.